src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -44,56 +44,58 @@ public static class PackageDownloader
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token))
+            // While CancellationTokenSource has a dispose, if we call it before the tasks using
+            // the token are done we risk having these tasks fail with a object disposed exception so
+            // this should not be in a using unless we want to block until all tasks are done/cancled
+            var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
```

Comment:
This is the change.  The rest of the modifications should be decreasing indention.